pkg/kclient/services.go

```diff
@@ -47,3 +49,35 @@ func (c *Client) ListServices(selector string) ([]corev1.Service, error) {
 	}
 	return serviceList.Items, nil
 }
+
+// DeleteService deletes the service with the given service name
+func (c *Client) DeleteService(serviceName string) error {
+	err := c.KubeClient.CoreV1().Services(c.Namespace).Delete(context.TODO(), serviceName, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetOneService retrieves the service with the given component and app name
+func (c *Client) GetOneService(componentName, appName string) (*corev1.Service, error) {
+	return c.GetOneServiceFromSelector(componentlabels.GetSelector(componentName, appName))
+}
+
+// GetOneServiceFromSelector returns the service object associated with the given selector.
+// An error is thrown when exactly one service is not found for the selector.
+func (c *Client) GetOneServiceFromSelector(selector string) (*corev1.Service, error) {
+	services, err := c.ListServices(selector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to get Deployments for the selector: %v", selector)
```

Comment:
As discussed with @mohammedzee1000 on the chat, it is recommended to use fmt.Errorf("unable to get Deployments for the selector %q: %w", selector, err)

Reference: https://blog.golang.org/go1.13-errors